File: src/Algorithm/LeetCode/LeetCode_209_Minimum_Size_Subarray_Sum/Solution.java
package Algorithm.LeetCode.LeetCode_209_Minimum_Size_Subarray_Sum;

class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        if (nums.length == 1) {
            return nums[0] >= target ? 1 : 0;
        }

        int result = Integer.MAX_VALUE;
        int i = 0, j = 0;
        int subtotal = 0;

        while (i < nums.length && j < nums.length) {
            subtotal += nums[j];
            j++;

            while (i < j && subtotal >= target) {
                result = Math.min(result, j - i);
                subtotal -= nums[i];
                i++;
            }
        }

        return result == Integer.MAX_VALUE ? 0 : result;
    }
}

File: src/Algorithm/LeetCode/LeetCode_209_Minimum_Size_Subarray_Sum/Solution.py
from typing import List

# https://www.programmercarl.com/0209.长度最小的子数组.html
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        i, sum, result = 0, nums[0], float('inf')

        for j in range(len(nums)):
            sum += nums[j]

            while sum >= target:
                result = min(result, j - i + 1)
                sum -= nums[i]
                i += 1

        return result if result != float('inf') else 0
